test/features/composer/presentation/composer_controller_test.dart

```diff
@@ -0,0 +1,852 @@
+          
+          const emailContent = 'some email content';
+          composerController?.onChangeTextEditorWeb(emailContent);
+            
+          const emailSubject = 'some email subject';
+          composerController?.subjectEmail.value = emailSubject;
+
+          final toRecipient = EmailAddress('to', 'to@linagora.com');
+          composerController?.listToEmailAddress = [toRecipient];
+
+          final ccRecipient = EmailAddress('cc', 'cc@linagora.com');
+          composerController?.listCcEmailAddress = [ccRecipient];
+
+          final bccRecipient = EmailAddress('bcc', 'bcc@linagora.com');
+          composerController?.listBccEmailAddress = [bccRecipient];
+
+          final selectedIdentity = Identity(id: IdentityId(Id('alice')));
+          composerController?.identitySelected.value = selectedIdentity;
+
+          final attachment = Attachment();
+          when(mockUploadController.attachmentsUploaded).thenReturn([attachment]);
+
+          composerController?.composerArguments.value = ComposerArguments();
+          when(
+            mockCreateNewAndSaveEmailToDraftsInteractor.execute(
+              createEmailRequest: anyNamed('createEmailRequest'),
+              cancelToken: anyNamed('cancelToken')))
+            .thenAnswer((_) => Stream.value(
+              Right(UpdateEmailDraftsSuccess(EmailId(Id('123'))))));
+
+          final savedEmailDraft = SavedEmailDraft(
+            content: emailContent,
+            subject: emailSubject,
+            toRecipients: {toRecipient},
+            ccRecipients: {ccRecipient},
+            bccRecipients: {bccRecipient},
+            identity: selectedIdentity,
+            attachments: [attachment],
+            hasReadReceipt: false
+          );
+
+          await tester.pumpWidget(WidgetFixtures.makeTestableWidget(
+            child: const Stack(children: [ComposerView()])));
+          await tester.pump();
+          
+          // act
+          final saveAsDraftButton = find.ancestor(
+            of: find.byType(InkWell),
+            matching: find.byWidgetPredicate(
+              (widget) => widget is TMailButtonWidget
+                && widget.icon == ImagePaths().icSaveToDraft));
+          await tester.tap(saveAsDraftButton);
+          await tester.pump();
+          await untilCalled(
+            mockCreateNewAndSaveEmailToDraftsInteractor.execute(
+              createEmailRequest: anyNamed('createEmailRequest'),
+              cancelToken: anyNamed('cancelToken')));
+          
+          // assert
+          expect(composerController?.savedEmailDraftHash, savedEmailDraft.hashCode);
+
+          // tear down
+          PlatformInfo.isTestingForWeb = false;
+        });
+      });
+
+      test(
+        'should update _savedEmailDraftHash '
+        'when there is a new view state '
+        'and the state is GetEmailContentSuccess '
+        'and emailActionType is editDraft',
+      () async {
```

Comment:
I found we call `_savedEmailDraftHash` in `_setUpRequestReadReceiptForDraftEmail` and `_restoreIdentityFromHeader`, can we need to make sure 2 calls of this still not make the hash changed?

Reply:
Test updated to verify that called twice without changing the hash.